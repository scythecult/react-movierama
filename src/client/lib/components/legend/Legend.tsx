import { Seat } from '../seat/Seat';
import styles from './styles.module.css';

export const Legend = () => {
  return (
    <ul className={styles.legend}>
      <li className={styles.legendItem}>
        <Seat isDisabled />
        <small>N/A</small>
      </li>
      <li className={styles.legendItem}>
        <Seat isDisabled isSelected />
        <small>Selected</small>
      </li>
      <li className={styles.legendItem}>
        <Seat isOccupied />
        <small>Occupied</small>
      </li>
    </ul>
  );
};
