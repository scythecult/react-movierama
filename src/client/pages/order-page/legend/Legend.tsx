import { SeatStateMap } from '@/client/lib/constants/common';
import type { SeatType } from '@/client/lib/types/OrderPageData';
import { Seat } from '../seats/seat/Seat';
import styles from './styles.module.css';

export type LegendProps = {
  seatTypes: SeatType[];
};

export const Legend = (props: LegendProps) => {
  const { seatTypes } = props;

  const seatTypeNodes = seatTypes.map((seatType) => {
    const { id, name, ticketTypes } = seatType;
    const [defaultTicketType] = ticketTypes;

    return (
      <li className={styles.legendItem} key={id} data-test-id="legend-item">
        <Seat className={styles.legendItemSeat} type={id} />
        <div className={styles.legendItemWrap} data-test-id="legend-item-wrap">
          <small className={styles.legendItemName} data-test-id="legend-item-name">
            {name}
          </small>
          <small className={styles.legendItemPrice} data-test-id="legend-item-price">
            {defaultTicketType.price} RUB
          </small>
        </div>
      </li>
    );
  });

  return (
    <ul className={styles.legend} data-test-id="legend">
      {seatTypeNodes}
      <li className={styles.legendItem} data-test-id="legend-item">
        <Seat className={styles.legendItemSeat} state={SeatStateMap.FREE} />
        <small className={styles.legendItemName} data-test-id="legend-item-name">
          N/A
        </small>
      </li>
      <li className={styles.legendItem} data-test-id="legend-item">
        <Seat className={styles.legendItemSeat} state={SeatStateMap.SELECTED} />
        <small className={styles.legendItemName} data-test-id="legend-item-name">
          Selected
        </small>
      </li>
      <li className={styles.legendItem} data-test-id="legend-item">
        <Seat className={styles.legendItemSeat} state={SeatStateMap.OCCUPIED} />
        <small className={styles.legendItemName} data-test-id="legend-item-name">
          Occupied
        </small>
      </li>
    </ul>
  );
};
