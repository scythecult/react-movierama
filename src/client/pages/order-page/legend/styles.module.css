.legend {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(90px, max-content));
    align-items: center;
    justify-content: center;
    width: 100%;
    gap: 16px;
    margin: 0;
    padding: 24px 0;
    color: var(--app-secondary-text);
    border-radius: var(--app-border-radius-m);
    background-color: var(--app-secondary-bg);
    list-style: none;
}

.legendItem {
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 6px;
}

.legendItemName {
    font-size: 14px;
    text-transform: capitalize;
    color: var(--app-primary-text);
}

.legendItemPrice {
    font-size: 14px;
}

.legendItemWrap {
    display: flex;
    flex-direction: column;
    gap: 2px;
}

.legendItemSeat {
    position: relative;
    pointer-events: none;
}

.legendItemSeat:hover {
    border-color: transparent;
    transform: scale(1);
    cursor: default;
}
