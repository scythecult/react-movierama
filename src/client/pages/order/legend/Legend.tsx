import { Seat } from '../seats/seat/Seat';
import styles from './styles.module.css';

export const Legend = () => {
  return (
    <ul className={styles.legend}>
      <li className={styles.legendItem}>
        <Seat className={styles.legendItemSeat} isDisabled />
        <small>N/A</small>
      </li>
      <li className={styles.legendItem}>
        <Seat className={styles.legendItemSeat} isDisabled isSelected />
        <small>Selected</small>
      </li>
      <li className={styles.legendItem}>
        <Seat className={styles.legendItemSeat} isOccupied />
        <small>Occupied</small>
      </li>
    </ul>
  );
};
