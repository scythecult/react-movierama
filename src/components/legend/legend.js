import React from "react";
import { Seat } from "../seat/seat";

import classes from "./legend.module.css";

const Legend = () => {
  return (
    <ul className={classes.legend}>
      <li className={classes["legend__item"]}>
        <Seat disabled={true} />
        <small>N/A</small>
      </li>
      <li className={classes["legend__item"]}>
        <Seat disabled={true} selected={true} />
        <small>Selected</small>
      </li>
      <li className={classes["legend__item"]}>
        <Seat occupied={true} />
        <small>Occupied</small>
      </li>
    </ul>
  );
};

export { Legend };
