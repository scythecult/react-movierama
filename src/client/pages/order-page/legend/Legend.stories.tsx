import type { Meta, StoryFn } from '@storybook/react-vite';
import { STATIC_SEAT_TYPES } from '../../../lib/utils/mocks';
import { Legend, type LegendProps } from './Legend';

export default {
  title: 'Pages/Legend',
  component: Legend,
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof Legend>;

const Template: StoryFn<LegendProps> = (props) => (
  <div style={{ padding: '2rem' }}>
    <Legend {...props} />
  </div>
);

export const Default = Template.bind(null);
Default.args = {
  seatTypes: STATIC_SEAT_TYPES,
};
