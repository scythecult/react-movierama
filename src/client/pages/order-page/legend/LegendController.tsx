import { useAppStore } from '../../../lib/contexts/app-store/useAppStore';
import { Legend } from './Legend';

export const LegendController = () => {
  const seatTypes = useAppStore((state) => state.seatTypes);

  return <Legend seatTypes={seatTypes} />;
};
