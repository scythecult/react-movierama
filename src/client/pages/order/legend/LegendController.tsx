import { useAppStore } from '@/client/lib/hooks/useAppStore';
import { Legend } from '@/client/pages/order/legend/Legend';

export const LegendController = () => {
  const seatTypes = useAppStore((state) => state.seatTypes);

  return (
    <Legend seatTypes={seatTypes} />
  );
};