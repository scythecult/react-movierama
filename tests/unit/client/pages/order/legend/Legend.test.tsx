import { render } from '@testing-library/react';
import { Legend } from '@/client/pages/order/legend/Legend';

describe('Legend', () => {
  test('should correspond default layout', () => {
    const result = render(<Legend />);

    expect(result.container).toMatchSnapshot();
  });
});
