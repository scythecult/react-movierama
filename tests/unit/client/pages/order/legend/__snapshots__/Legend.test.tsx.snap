// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Legend > should correspond default layout 1`] = `
<div>
  <ul
    class="_legend_28be33"
  >
    <li
      class="_legendItem_28be33"
    >
      <button
        class="_seat_aa3898 _legendItemSeat_28be33 _disabled_aa3898"
        type="button"
      />
      <small>
        N/A
      </small>
    </li>
    <li
      class="_legendItem_28be33"
    >
      <button
        class="_seat_aa3898 _legendItemSeat_28be33 _selected_aa3898 _disabled_aa3898"
        type="button"
      />
      <small>
        Selected
      </small>
    </li>
    <li
      class="_legendItem_28be33"
    >
      <button
        class="_seat_aa3898 _legendItemSeat_28be33 _occupied_aa3898"
        disabled=""
        type="button"
      />
      <small>
        Occupied
      </small>
    </li>
  </ul>
</div>
`;
